### PR TITLE
#167165075 article bookmark

### PR DESCRIPTION
**Pull request by PelumiAlesh**
### What does this PR do?
Enables user to bookmark an article 
Enables user to view an all bookmarks
#### Description of Task to be completed?
Bookmarks (1)

- Build endpoint for user to bookmark an article
- Build an endpoint for user to view all the article that has been bookmarked
#### How should this be manually tested?
Clone the repo and run `npm i` Then 
>>make a `POST` request to `api/v1/articles/:id/bookmark` to bookmark an article with the articles ID
>> make a `GET` request to `api/v1/users/bookmarks` to get all the articles the user bookmarked
#### What are the relevant pivotal tracker stories?
#167165075 
#### Screenshots (if appropriate)
Screenshot 2019-08-07 at 5 39 17 AM
Screenshot 2019-08-07 at 5 38 08 AM

### CODE REVIEW

**Review by Tunji17 — APPROVED** [review]
LGTM

**Review by VictorMaria — APPROVED** [review]

File: routes/userRoutes.js [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
-  getAuthorOwnArticles
+  getAuthorOwnArticles,
+  getAllBookmark
 } = ArticleController;
 const { verifyToken } = AuthMiddleware;
 
 const router = Router();
 
 router.get('/articles', verifyToken, getAuthorOwnArticles);
+router.get('/:id/bookmarks', verifyToken, Validation.validateUserId, getAllBookmark);
```
Comment (ocranbillions) [inline routes/userRoutes.js:15]: I see you've implemented the feedback. Good job

**Review by ocranbillions — APPROVED** [review]
LGTM

File: controllers/ArticleController.js [hunk 2]
```diff
@@ -441,4 +441,72 @@ export default class ArticleController {
...
+   * @param {Object} res express respond object
+   * @param {function} next
+   * @returns {object} All the articles bookmarked by the user
+   */
+  static async getAllBookmark(req, res, next) {
+    try {
+      const { id } = req.user;
+      const { id: userId } = req.params;
+      if (id !== userId) {
+        return errorResponse(res, 401, 'You are not allowed to view this user\'s bookmarks');
```
Comment (mazma1) [inline controllers/ArticleController.js]: Is `401` the right status code for this?

Comment (PelumiAlesh) [reply controllers/ArticleController.js]: Yes, that means the user is inputing a valid `id`, but the `id` belongs to someone else. So he/she is not authorised.

Comment (mazma1) [reply controllers/ArticleController.js]: In that case, it is not an authentication error. The user does not have the permission to view another user's bookmarks. Is that still a `401`?

Comment (PelumiAlesh) [reply controllers/ArticleController.js]: Oh yeah, that's true. I would change it now

Comment (PelumiAlesh) [reply controllers/ArticleController.js]: Fixed @mazma1

**Pull request merged**